core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -121,7 +121,7 @@ private[spark] class Worker(
   val shuffleService = new StandaloneWorkerShuffleService(conf, securityMgr)
 
   val publicAddress = {
```

Comment:
It just occurred to me that this might not be the right `publicAddress` to use: I think that we allow the `--host` flag to be passed when launching the worker, in which case this will default to the value of that flag, whereas the UI defaults to Utils.localHostname(), so there's still a slight chance for a discrepancy here.  Maybe I can grab the bound host out of the web UI instance instead, just to be safe.